resources/[soz]/soz-core/src/nui/components/Shop/BarberShopMenu.tsx

```diff
@@ -0,0 +1,396 @@
+import { PlayerData } from '@public/shared/player';
+import { BarberConfiguration, BarberShopColors, BarberShopContent } from '@public/shared/shop';
+import { FunctionComponent, useState } from 'react';
+
+import { NuiEvent } from '../../../shared/event';
+import { MenuType } from '../../../shared/nui/menu';
+import { fetchNui } from '../../fetch';
+import {
+    MainMenu,
+    Menu,
+    MenuContent,
+    MenuItemButton,
+    MenuItemCheckbox,
+    MenuItemSelect,
+    MenuItemSelectOption,
+    MenuItemSelectOptionColor,
+    MenuTitle,
+} from '../Styleguide/Menu';
+
+type MenuBarberShopStateProps = {
+    data: {
+        shop_content: BarberShopContent;
+        player_data: PlayerData;
+        shop_colors: BarberShopColors;
+    };
+};
+
+export const BarberShopMenu: FunctionComponent<MenuBarberShopStateProps> = ({ data }) => {
+    const banner = 'https://nui-img/soz/menu_shop_barber';
+    const content = data.shop_content[data.player_data.skin.Model.Hash];
+    const [configuration, setConfiguration] = useState<Record<string, BarberConfiguration>>({});
+    const [totalConfiguration, setTotalConfiguration] = useState<BarberConfiguration>({ Hair: {}, Makeup: {} });
+
+    const updateConfiguration = (category: string, overlay: string, field: string, value: number | boolean) => {
+        configuration[category] = configuration[category] || { Hair: {}, Makeup: {} };
+        configuration[category][overlay][field] = value;
+        setConfiguration(configuration);
+        totalConfiguration[overlay][field] = value;
+        setTotalConfiguration(totalConfiguration);
+        fetchNui(NuiEvent.BarberShopPreview, totalConfiguration);
+    };
+
+    return (
+        <Menu type={MenuType.BarberShop}>
+            <MainMenu>
+                <MenuTitle banner={banner}>Coiffeur</MenuTitle>
+                <MenuContent>
+                    {content.map(cat => {
+                        const title = <MenuTitle>{cat.label}</MenuTitle>;
+                        let core;
+                        const validateButton = (
+                            <MenuItemButton
+                                onConfirm={async () => {
+                                    await fetchNui(NuiEvent.BarberShopBuy, {
+                                        configuration: configuration[cat.category],
+                                        price: cat.price,
+                                        overlay: cat.overlay,
+                                    });
+                                }}
+                            >
+                                <div className="flex justify-between items-center">
+                                    <span>Valider les modifications</span>
+                                    <span className="mr-1">${cat.price}</span>
+                                </div>
+                            </MenuItemButton>
+                        );
+
+                        switch (cat.category) {
```

Comment:
Please avoid switch case in JSX. You should use other small components.

Reply:
Done. Reworked BarberShopMenu